internal/teststep/config.go

```diff
@@ -45,6 +45,7 @@ type Config interface {
 	HasProviderBlock(context.Context) (bool, error)
 	HasTerraformBlock(context.Context) (bool, error)
 	Write(context.Context, string) error
+	Append(context.Context, string) Config
```

Comment:
This feels simpler than `teststep.Configuration(teststep.PrepareConfigurationRequest({...}))`.

The thought is: given a `Config` value, produce a new `Config` value that includes the appended generated config.

And then each concrete type implements how to write its "base" config and the appended generated config.

For a string: concatenate.

For a file: append to the file.

For a directory: add a file to the directory.

Reply:
Very clean, like it 👍🏻, and it looks like we just hold that in a buffer until we read the data which is good.

Reply:
💯